dubbo-common/src/main/java/org/apache/dubbo/common/URL.java

```diff
@@ -477,7 +484,7 @@ public int getPort(int defaultPort) {
     }
 
     public String getAddress() {
-        return port <= 0 ? host : host + ":" + port;
+        return address;

```

Comment:
Should we check if `address` is empty before return? Or compute address like before when address is empty?

Reply:
`new URL(...)` is not the only way of allocating URL instance, if URL instance is allocated in other way, for example deserialization, the address property would be left empty.